### PR TITLE
`average` implementation and coverage

### LINKED ISSUE #13: [Feature]: `average` function
This function should return the average of two numbers. 

Check [Solidity Math Library](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/math/Math.sol) for the reference.

### PR DESCRIPTION
**Pull request by ericnordelo**
Fixes #13 

Move Coverage: 96.32

## Release Notes

* **New Features**
  * Added `average()` function for computing arithmetic mean across all unsigned integer types with configurable rounding modes (down, up, nearest).

* **Tests**
  * Added comprehensive test coverage for the new average function, including various rounding modes and edge cases with large inputs.
  * Reorganized existing test structure with improved naming conventions for clarity.

* **Chores**
  * Minor formatting updates to configuration files.

### CODE REVIEW

File: math/core/sources/macros.move [hunk 1]
```diff
@@ -41,6 +41,31 @@ public(package) macro fun mul_div<$Int>(
...
+    let a_u256 = ($a as u256);
+    let b_u256 = ($b as u256);
+    let rounding_mode = $rounding_mode;
+
+    let mut lower = a_u256;
+    let mut upper = b_u256;
+    if (lower > upper) {
+        lower = b_u256;
+        upper = a_u256;
+    };
```
[the remark this answers is not in the document]
Comment (ericnordelo) [reply math/core/sources/macros.move]: We should benefit clarity when there is no significant improvement, and this is one of those cases to me.

[the remark this answers is not in the document]
Comment (immrsd) [reply math/core/sources/macros.move]: Okay, in that case I think we're good to merge it

File: math/core/tests/macros_tests.move [hunk 2]
```diff
@@ -147,3 +149,32 @@ fun macro_uses_wide_path_for_large_inputs() {
...
+    assert_eq!(nearest, 2u16);
+
+    let reversed = macros::average!(7u32, 4u32, rounding::down());
+    assert_eq!(reversed, 5u32);
+}
+
+#[test]
+fun average_handles_large_inputs() {
+    let max = std::u256::max_value!();
+    let almost = max - 1u256;
```
[the remark this answers is not in the document]
Comment (immrsd) [reply math/core/tests/macros_tests.move]: I see that one marked as resolved, but no actual change in the code

File: math/core/sources/macros.move (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericnordelo) [reply math/core/sources/macros.move]: Remeber we need you approval on the PR for that to happen.
